heat/core/dndarray.py

```diff
@@ -818,28 +871,46 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
                 if isinstance(key_stop, torch.Tensor):
                     key_stop = key_stop.item()
                 key[self.split] = slice(key_start, key_stop, key_step)
-                arr = self.__array[tuple(key)]
-
-        elif isinstance(key[self.split], int):
-            # if there is an integer in the key along the split axis, adjust it and then get `arr`
+                lout[new_split] = (
+                    math.ceil((key_stop - key_start) / key_step)
+                    if key_step is not None
+                    else key_stop - key_start
+                )
+                arr = self.__array[tuple(key)].reshape(lout)
+            else:
+                lout[new_split] = 0
+                arr = torch.empty(lout, dtype=self.__array.dtype, device=self.__array.device)
+
+        elif (
+            isinstance(key[self.split], int)
+            or isinstance(key[self.split], (list, torch.Tensor, DNDarray, np.ndarray))
+            and len(key[self.split]) == 1
+        ):
+            # slicing one item along split axis:
             key = list(key)
-            key[self.split] = (
-                key[self.split] + self.gshape[self.split]
-                if key[self.split] < 0
-                else key[self.split]
-            )
-            if key[self.split] in range(chunk_start, chunk_end):
-                key[self.split] = key[self.split] - chunk_start
-                arr = self.__array[tuple(key)]
+            if isinstance(key[self.split], list):
+                key[self.split] = key[self.split].pop()
+            elif isinstance(key[self.split], (torch.Tensor, DNDarray, np.ndarray)):
+                key[self.split] = key[self.split].item()
+            # translate negative index
+            if key[self.split] < 0:
+                key[self.split] += self.gshape[self.split]
+
+            active_rank = torch.where(key[self.split] >= chunk_starts)[0][-1].item()
+            # slice `self` on `active_rank`, allocate `arr` on all other ranks in preparation for Bcast
+            if rank == active_rank:
+                key[self.split] -= chunk_start.item()
+                arr = self.__array[tuple(key)].reshape(tuple(lout))
+            else:
+                arr = torch.empty(tuple(lout), dtype=self.larray.dtype, device=self.larray.device)
+            # broadcast result
+            self.comm.Barrier()
```

Comment:
needed?

Reply:
thanks, not really, was left over from tests.